src/plots/gl3d/layout/defaults.js

```diff
@@ -97,10 +97,19 @@ function handleGl3dDefaults(sceneLayoutIn, sceneLayoutOut, coerce, opts) {
         sceneLayoutIn.aspectmode = sceneLayoutOut.aspectmode;
     }
 
+    var fullGl3dData = [];
+    for(var i = 0; i < opts.fullData.length; i++) {
+        if(opts.fullData[i].scene) {
+            fullGl3dData.push(
+                opts.fullData[i]
+            );
+        }
+    }
+
     supplyGl3dAxisLayoutDefaults(sceneLayoutIn, sceneLayoutOut, {
         font: opts.font,
         scene: opts.id,
-        data: opts.fullData,
+        data: fullGl3dData,
```

Comment:
Nicely done, but let's :recycle: 

https://github.com/plotly/plotly.js/blob/0078626c6ef3809898cfceab76e8ba227897e046/src/plots/get_data.js#L89-L126

Reply:
@etpinard The try to reuse the getSubPlotData function was not successful.
Could I revert this change?

Reply:
Many thanks @etpinard for the help figuring this out!
Now the tests are all passed.